File: files/200-299/276/CH-276.py
import pandas as pd

path = "200-299/276/CH-276 Value COnversion.xlsx"
input = pd.read_excel(path, usecols="B:E", nrows=6, skiprows=1)
test = pd.read_excel(path, usecols="F", nrows=6, skiprows=1).astype({'Converted Value': str})

def convert_base(x, from_base, to_base):
    n = int(x, from_base)
    if to_base == 2: return bin(n)[2:]
    if to_base == 8: return oct(n)[2:]
    if to_base == 10: return str(n)
    if to_base == 16: return hex(n)[2:].upper()

input['Result'] = input.apply(
    lambda row: convert_base(str(row['Number']), int(row['From Base']), int(row['To Base'])),
    axis=1
)

print(input['Result'].equals(test['Converted Value']))

File: files/200-299/276/CH-276.R
library(tidyverse)
library(readxl)
library(gmp)

path = "files/200-299/276/CH-276 Value COnversion.xlsx"
input = read_excel(path, range = "B2:E8")
test  = read_excel(path, range = "F2:F8")

convert_base = function(x, from, to) {
  pmap_chr(list(x, from, to), \(x, from, to) {
    n = strtoi(x, base = from)
    switch(as.character(to),
      "2"  = as.character(as.bigz(n), b=2),
      "8"  = as.character(as.octmode(n)),
      "10" = as.character(n),
      "16" = str_to_upper(as.character(as.hexmode(n)))
    )
  })
}

result = input %>%
  mutate(Answer = convert_base(Number, `From Base`, `To Base`)) 

all.equal(result$Answer, test$`Converted Value`)
# > [1] TRUE